src/components/navigation-header/components/product-page-content/utils/get-nav-items.ts

```diff
@@ -0,0 +1,57 @@
+import { NavigationHeaderIcon } from 'components/navigation-header/types'
+import { NavItem } from './types'
+import { ProductData, RootDocsPath } from 'types/products'
+
+/**
+ * Given current product data,
+ * Return an array of NavItems to render in the top navigation bar.
+ *
+ * Note that these items are only shown on larger viewports.
+ * On smaller viewports, a separate mobile menu is shown, which
+ * does not use this exact function to generate nav items,
+ * so may not show the same set of items.
+ */
+export function getNavItems(currentProduct: ProductData): NavItem[] {
+	/**
+	 * Define a common set of base nav items
+	 */
+	const items: NavItem[] = [
+		{ type: 'link', label: 'Home', url: `/${currentProduct.slug}` },
+		{
+			type: 'submenu',
+			label: 'Documentation',
+			iconColorTheme: currentProduct.slug,
+			items: currentProduct.rootDocsPaths.map((rootDocsPath: RootDocsPath) => {
+				return {
+					icon: rootDocsPath.iconName as NavigationHeaderIcon,
+					label: rootDocsPath.name,
+					path: `/${currentProduct.slug}/${rootDocsPath.path}`,
+				}
+			}),
+		},
+		{
+			type: 'link',
+			label: 'Tutorials',
+			url: `/${currentProduct.slug}/tutorials`,
+		},
+		{
+			type: 'link',
+			label: 'Install',
+			url: `/${currentProduct.slug}/tutorials`,
+		},
+	]
+	/**
+	 * For Terraform, add a "Registry" item
+	 */
+	if (currentProduct.slug == 'terraform') {
```

Comment:
It felt a bit odd to push an extra link based on a product slug, rather than for example managing nav data in a JSON file as authorable content. But I think this aligns with our current intent of keeping as much consistency as possible between products.